docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
+One particular feature of certain table documents is that they can be very
 large, since they may contain huge quantities of data; hence there should be a
 way in which they can be processed by chunks (i.e. as streaming objects),
 without the need to hold all their contents at once.
 
-As all source document model types, a tabular document is considered as a
+As all source document model types, a table document is considered as a
 document header, plus a sequence of document chunks.
 
-The document header corresponds to the specification in the previous section,
-with one modification in the `context` element: in addition to the already
-defined `document` and `dataset` contexts, there may also be a `column`
-context, providing specific information for each of the columns in the document
-(considered in sequential order). It could contain up to two subelements:
+The document header corresponds to the already defined specification, with one
+modification: in addition to the already defined `document` and `dataset`
+sections, there may also be a `column` section, providing specific information
+for each of the columns in the document (considered in sequential order). It
+could contain up to two subelements:
 * `name`: the column names, as a list of text strings
 * `description`: a description for each column, also as a list of text strings
 
-Beyond the header, each chunk contains a small dictionary with 2 or 3 elements:
-* `id`: an arbitrary string that should be unique per document. Its mission is
-  to make it easier later on to map detected PII instances to the chunk they
-  are part of
-* `table`: a 2D array holding the tabular data for this hunk, as a 2-level
-  list of first rows and, inside each row, columns. The contents of each table 
-  position (a table cell) is an arbitrary text string
-* `context`: elements that help to provide additional information to help in
-  the detection and decision process. There can be three types of context, all
-  of them optional:
-   - _global_: it will be a link/transposition of the context in the document
-     header, as described above
-   - _before_: a small section of data from the chunk preceding this one,
-     containing the closest rows to this chunk
-   - _after_: a small section of data from the chunk following this one,
-     containing the closest rows to this chunk
 
-Note that the context element of the chunk is a logical one, and need not be
-present in a static representation of the chunk, or when sent or streamed,
-since it would repeat information unnecessarily. Instead, it would be generated
-on the fly by an appropriate PII processing module, so that processing
-elements down the line have direct access to that context when processing a
-chunk, even when in streaming mode.
+#### Structural iteration
 
+* A document is considered as a sequence of _rows_
+* Each row contains a list of column values, each column value is a string with
```

Comment:
There could be some complexities here. For instance, if your table has thousands of columns. Also, there are different tabular data representations (parquet vs. avro for example). This would also influence the way data is read from the store.

Probably not at this point, but we should consider supporting the processing of a sequence of rows, and a sequence per column.

Reply:
good points

Reply:
Let's keep it aside for now